tsdb/head.go

```diff
@@ -1378,9 +1417,30 @@ func (a *headAppender) log() error {
 			return errors.Wrap(err, "log samples")
 		}
 	}
+	if len(a.exemplars) > 0 {
+		rec = enc.Exemplars(exemplarsForEncoding(a.exemplars), buf)
```

Comment:
Is some refactoring possible to skip `exemplarsForEncoding` step? I am not sure if it would be required, just guessing this might be some additional unnecessary allocations.